x-pack/plugins/infra/public/components/saved_views/manage_views_flyout.tsx

```diff
@@ -22,107 +22,85 @@ import {
+import { UseMetricsExplorerViewsResult } from '../../hooks/use_metrics_explorer_views';
 
-interface Props<ViewState> {
-  views: Array<SavedView<ViewState>>;
+type View = InventoryView | MetricsExplorerView;
+type UseViewResult = UseInventoryViewsResult | UseMetricsExplorerViewsResult;
+
+export interface ManageViewsFlyoutProps {
+  views: UseViewResult['views'];
   loading: boolean;
-  sourceIsLoading: boolean;
   onClose(): void;
-  onMakeDefaultView(id: string): void;
-  setView(viewState: ViewState): void;
-  onDeleteView(id: string): void;
+  onMakeDefaultView: UseViewResult['setDefaultViewById'];
+  onSwitchView: UseViewResult['switchViewById'];
+  onDeleteView: UseViewResult['deleteViewById'];
 }
 
 interface DeleteConfimationProps {
   isDisabled?: boolean;
   onConfirm(): void;
 }
 
-const DeleteConfimation = ({ isDisabled, onConfirm }: DeleteConfimationProps) => {
-  const [isConfirmVisible, toggleVisibility] = useToggle(false);
-
-  return isConfirmVisible ? (
-    <EuiFlexGroup>
-      <EuiButtonEmpty onClick={toggleVisibility} data-test-subj="hideConfirm">
-        <FormattedMessage defaultMessage="cancel" id="xpack.infra.waffle.savedViews.cancel" />
-      </EuiButtonEmpty>
-      <EuiButton
-        disabled={isDisabled}
-        fill={true}
-        iconType="trash"
-        color="danger"
-        onClick={onConfirm}
-        data-test-subj="showConfirm"
-      >
-        <FormattedMessage
-          defaultMessage="Delete view?"
-          id="xpack.infra.openView.actionNames.deleteConfirmation"
-        />
-      </EuiButton>
-    </EuiFlexGroup>
-  ) : (
-    <EuiButtonEmpty
-      data-test-subj="infraDeleteConfimationButton"
-      iconType="trash"
-      color="danger"
-      onClick={toggleVisibility}
-    />
-  );
+const searchConfig = {
+  box: { incremental: true },
 };
 
-export function SavedViewManageViewsFlyout<ViewState>({
+export function ManageViewsFlyout({
   onClose,
-  views,
-  setView,
+  views = [],
+  onSwitchView,
   onMakeDefaultView,
   onDeleteView,
   loading,
-  sourceIsLoading,
-}: Props<ViewState>) {
-  const [inProgressView, setInProgressView] = useState<string | null>(null);
+}: ManageViewsFlyoutProps) {
+  // Add name as top level property to allow in memory search
+  const namedViews = useMemo(() => views.map(addOwnName), [views]);
 
-  const renderName = (name: string, item: SavedView<ViewState>) => (
+  const renderName = (name: string, item: View) => (
     <EuiButtonEmpty
       key={item.id}
       data-test-subj="infraRenderNameButton"
       onClick={() => {
-        setView(item);
+        onSwitchView(item.id);
         onClose();
       }}
     >
       {name}
     </EuiButtonEmpty>
   );
 
-  const renderDeleteAction = (item: SavedView<ViewState>) => {
+  const renderDeleteAction = (item: View) => {
     return (
       <DeleteConfimation
         key={item.id}
-        isDisabled={item.isDefault}
+        isDisabled={item.attributes.isDefault}
```

Comment:
Q: Should we do this check earlier and disable the delete icon button? Maybe showing a tooltip that the default view can't be deleted if it's not clear there 🤔  When I tested I saw that I can still click it the delete icon and after that I see a disabled button:
 

https://user-images.githubusercontent.com/14139027/235186694-296dda5a-0b1e-4446-8835-e940eb5ceef1.mov

Reply:
I spoke with @kkurstak in advance about this and I agree we could make a lot of improvements on the UX for saved views!
This work is however intended to only migrate the implementation for the saved views feature, we'll have the chance to further improve its usage when this work is finalized.